api_calls_string.py

```diff
@@ -0,0 +1,455 @@
+                        
+                    
+    # Catching exceptions #
+    except json.JSONDecodeError as e:
+        print(f"A JSONDecodeError occurred: {str(e)}")
+    except http.client.HTTPException as e:
+        print(f"An exception occurred: {str(e)}")
+    except Exception as e:
+        print(f"An exception occurred: {str(e)}")
+    return None
+
+# checks if there's any games between hours (est military time)
+#   and returns all
+def game_in_time(hour_start, hour_end):
+    # makes sure there's a schedule to read from
+    get_schedule()
+
+    #opens the schedule file
+    output = ""
+    with open(schedule_file, 'r') as file:
+        games = csv.reader(file)
+        first = True
+        for game in games:
+            if first: # gets past the first line
+                first = False
+            else:
+                # gets the date of the game
+                time = game[0]
+                game_year = int(time[0:4])
+                game_month = int(time[5:7])
+                game_day = int(time[8:10])
+                #checks the date
+                if game_year == year:
+                    
+                    if game_month == month:
+                        
+                        if game_day == day:
+                            # checks the time
+                            game_hour = int(time[11:13])
+
+                            if (game_hour >= hour_start) and (game_hour <= hour_end):  # game found
+                                output += f"{game[1]}vs{game[2]}\n"
+                        #if we've gone past the current date, stop checking
+                        elif game_day > day:
+                            return output
+                            
+
+    return output # just in case there are no more games in the season
+
+# Makes a csv file containing player names and ID                       
+def player_id_list():
+    #if player_list exists, exit
+    if os.path.isfile(player_list):
+        return None
+    else:
+        # makes sure teams list is present
+        get_teams()
+        
+        # creates array with team IDs
+        team_IDs = []
+        with open(teams_file, 'r') as file:
+            teams = csv.reader(file)
+            line = 0
+            
+            for team in teams:
+                
+                #if at first row, do nothing
+                if line != 0:
+                    team_IDs.append(team[0])
+                        
+                line += 1
+
+
+        # makes the player list
+        players = open(player_list,"w", newline = "")
+        writer = csv.writer(players)
+        writer.writerow(["Player_Name", "Player_ID", "Team_ID"])
+
+
+
+        # goes through each team, and add players to the file
+        try:
+            for team_id in team_IDs:
+
+                time.sleep(5) #This makes it take a while, but it stops the api from complaining about too many calls

```

Comment:
You could reduce it to time.sleep(2) as its needs minimum of one second per call